boredInTheHouse.js

```diff
@@ -0,0 +1,63 @@
+//creates instance of audio context
+const AudioContext = window.AudioContext || window.webkitAudioContext
+const audioContext = new AudioContext()
+
+//gets audio element
+const audioElement = document.querySelector('audio')
+
+// passes audio element into the audio context
+const track = audioContext.createMediaElementSource(audioElement)
+
+// connects element to contect destination
+track.connect(audioContext.destination)
+
+// creates play button
+const playButton = document.querySelector('button')
+
+playButton.addEventListener('click', function() { // allows play or pause
+  if (audioContext.state === 'suspended') {
+      audioContext.resume();
+  }
+  if (this.dataset.playing === 'false') {
+      audioElement.play();
+      this.dataset.playing = 'true';
+  } else if (this.dataset.playing === 'true') {
+      audioElement.pause();
+      this.dataset.playing = 'false';
+  }
+
+}, false)
```

Comment:
This recommendation is for this piece of code but applies to all four similar cases:
For clarity and organization, it would be better if you extract all four **_addEventListener_** callbacks to stand-alone functions and then pass those to **_addEventListener_**, _AND_ move all four calls of **_addEventListener_** to the bottom of the code. Something like:

```suggestion
function playPause() { // allows play or pause
  if (audioContext.state === 'suspended') {
      audioContext.resume();
  }
  switch (this.dataset.playing) {
    case 'false':
      audioElement.play();
      this.dataset.playing = 'true';
      break;
    case 'true':
      audioElement.pause();
      this.dataset.playing = 'false';
  }
}
.
.
.
// Bottom of code
playButton.addEventListener('click', playPause);
```

This way you are much more explicit on what the action is doing, and grouping all **_addEventListener_** 's in one place.

Better yet: Group all four **_addEventListener_** calls in one function named **_initialization_**, together with other initialization related calls, and call it at the bottom. This, of course, requires more function definitions, but you would be grouping related tasks, and anyone reading the code would easily follow your train of thought.